src/content/docs/en/guides/cms/flotiq.mdx

```diff
@@ -0,0 +1,237 @@
+    props: post
+  })) || []
+}
+---
+```
+
+Next, add the logic to display a single post:
+
+```astro title="src/pages/posts/[slug].astro" ins={12-20}
+---
+import type { Blogpost } from "flotiq-api-ts";
+import { flotiq } from "../../lib/flotiq";
+
+export async function getStaticPaths() {
+  const posts = await flotiq.BlogPostAPI.list();
+  return posts.data?.map((post) => ({
+    params: { slug: post.slug },
+    props: post
+  })) || []
+}
+const post: Blogpost = Astro.props;
+---
+<html lang="en">
+  <title>{post.title}</title>
+  <body>
+    <h1>{post.title}</h1>
+    <div set:html={post.content}/>
+  </body>
+</html>
+```
+
+Visit `http://localhost:4321` and click on a blog post. You should be redirected to the individual post's page.
+
+#### Server-Side Rendering
+
+If you are using [SSR](/en/guides/server-side-rendering/) mode, you will need to fetch a single post based on its `slug`.
+
+Create a new file `[slug].astro` in the `/src/pages/posts/` directory. Fetch the post by its `slug` field:
+
+```astro title="src/pages/posts/[slug].astro"
+---
+import type { Blogpost } from "flotiq-api-ts";
+import { flotiq } from "../../lib/flotiq";
+
+const { slug } = Astro.params;
+let post: Blogpost;
+
+const blogpostList = await flotiq.BlogPostAPI.list({
+  filters: JSON.stringify({
+    slug: {
+      type: 'equals',
+      filter: slug,
+    }
+  }),
+  limit: 1
+});
+
+if (blogpostList.data?.[0]) {
+  post = blogpostList.data[0]
+} else {
+  return Astro.redirect('/404');
+}
+---
+```
+
+Next, add the logic to display a single post:
+
+```astro title="src/pages/posts/[slug].astro" ins={24-30}
+---
+import type { Blogpost } from "flotiq-api-ts";
+import { flotiq } from "../../lib/flotiq";
+
+const { slug } = Astro.params;
+let post: Blogpost;
+
+const blogpostList = await flotiq.BlogPostAPI.list({
+  filters: JSON.stringify({
+    slug: {
+      type: 'equals',
+      filter: slug,
+    }
+  }),
+  limit: 1
+});
+
+if (blogpostList.data?.[0]) {
+  post = blogpostList.data[0]
+} else {
+  return Astro.redirect('/404');
+}
+---
+<html lang="en">
+  <title>{post.title}</title>
+  <body>
+    <h1>{post.title}</h1>
+    <div set:html={post.content}/>
+  </body>
+</html>
+```
+
+Go to `http://localhost:4321` and click on the selected post. You should see the individual post's page.
+
+## Publishing Your Site
+
+To deploy your website, visit Astro's [deployment guides](/en/guides/deploy/) and follow the instructions for your preferred hosting provider.
+
+### Rebuild on Flotiq Changes
+
+If you're using Static Site Generation (SSG) mode (the default in Astro), the typical approach is to send a webhook from your Headless CMS to the hosting provider to trigger a rebuild.
+
+In Flotiq, simply add a new Webhook, define which Content Type and events it should trigger on, and configure it accordingly. You can find more information in the [Flotiq Webhooks documentation](https://flotiq.com/docs/panel/webhooks/async-co-webhook/).
+
```

Comment:
zabrakło mi informacji jak postąpić kiedy chcę zmienić ctd (nowe FlotiqApi nie jest ładowane automatycznie)

Reply:
Good catch; I'll add: 
```

## Refreshing the SDK After Content Type Changes

When using the Flotiq TypeScript SDK (`flotiq-api-ts`), all your data types are accurately mapped into the Astro project.

If you make changes to the structure of your content types (such as adding a new field or modifying an existing one), you’ll need to refresh the SDK to ensure that your project reflects the latest model updates.

To do this, simply rerun the following command:

yarn add flotiq-api-ts


This will update the SDK, aligning object types, fields, and API methods with your current data model.
```